chromeapp/app/scripts/dnssd/dns-controller.js

```diff
@@ -0,0 +1,585 @@
+/*jshint esnext:true*/
+/* globals Promise */
+'use strict';
+
+var chromeUdp = require('../chrome-apis/udp');
+var dnsUtil = require('./dns-util');
+var dnsPacket = require('./dns-packet');
+var byteArray = require('./byte-array');
+var dnsCodes = require('./dns-codes');
+var qSection = require('./question-section');
+
+/**
+ * This module maintains DNS state and serves as the DNS server. It is
+ * responsible for issuing DNS requests.
+ */
+
+var DNSSD_MULTICAST_GROUP = '224.0.0.251';
+var DNSSD_PORT = 53531;
+var DNSSD_SERVICE_NAME = '_services._dns-sd._udp.local';
+
+/** True if the service has started. */
+var started = false;
+
+exports.DNSSD_MULTICAST_GROUP = DNSSD_MULTICAST_GROUP;
+exports.DNSSD_PORT = DNSSD_PORT;
+exports.DNSSD_SERVICE_NAME = DNSSD_SERVICE_NAME;
+
+/**
+ * These are the records owned by this module. They are maintained in an object
+ * of domain name to array of records, e.g. { 'www.example.com': [Object,
+ * Object, Object], 'www.foo.com': [Object] }.
+ */
+var records = {};
+
+var onReceiveCallbacks = [];
+
+/**
+ * The IPv4 interfaces for this machine, cached to provide synchronous calls.
+ */
+var ipv4Interfaces = [];
+
+/**
+ * Returns all records known to this module.
+ *
+ * @return {Array<resource record>} all the resource records known to this
+ * module
+ */
+exports.getRecords = function() {
+  return records;
+};
+
+/**
+ * Returns all the callbacks currently registered to be invoked with incoming
+ * packets.
+ *
+ * @return {Array<function>} all the onReceive callbacks that have been
+ * registered
+ */
+exports.getOnReceiveCallbacks = function() {
+  return onReceiveCallbacks;
+};
+
+/**
+ * The socket used for accessing the network. Object of type
+ * chromeUdp.ChromeUdpSocket.
+ */
+exports.socket = null;
```

Comment:
Dumb question: Why not provide a getter for these two variables as well? Is there a reason to do that for records and onReceiveCallbacks but not socket and socketInfo?